### PR TITLE
Add the 'postbooting' status during the OS provisioining

### PR DESCRIPTION
**Pull request by robin2008**
A new status 'postbooting' to indicate xCAT start to handle postboot tasks (#4346)

 - for diskful, it will be set when leaving 'booting' stage
 - for diskless, it will be set when leaving 'netbooting' stage
 - for diskful reboot, it will set when (site.runpostscript=1)
 - modify the document accordingly
 - also modify GlobalDef.pm, but it won't have impact as the status will not be in nodeset operation.

### CODE REVIEW

**Comment by robin2008** [comment]
UT:
Diskful provisioning
```
Nov 23 20:57:56 briggs01 xcat: mid05tor12cn15 status: powering-off statustime: 11-23-2017 20:57:56
Nov 23 20:58:12 briggs01 xcat: mid05tor12cn15 status: powering-on statustime: 11-23-2017 20:58:12
Nov 23 21:01:27 briggs01 xcat: mid05tor12cn15 status: installing statustime: 11-23-2017 21:01:27
Nov 23 21:11:40 briggs01 xcat: mid05tor12cn15 status: booting statustime: 11-23-2017 21:11:40
Nov 23 21:15:10 briggs01 xcat: mid05tor12cn15 status: postbooting statustime: 11-23-2017 21:15:10
Nov 23 21:15:11 briggs01 xcat: mid05tor12cn15 status: booted statustime: 11-23-2017 21:15:11
```

Diskful reboot, with `RUNBOOTSCRIPTS=i` in `site`.
```
Nov 23 23:57:00 briggs01 xcat: mid05tor12cn15 status: powering-off statustime: 11-23-2017 23:57:00
Nov 24 00:00:11 briggs01 xcat: mid05tor12cn15 status: postbooting statustime: 11-24-2017 00:00:11
Nov 24 00:00:45 briggs01 xcat: mid05tor12cn15 status: booted statustime: 11-24-2017 00:00:45
```
Stateless provisioning:
```
Nov 23 21:04:52 briggs01 xcat: mid05tor12cn16 status: powering-off statustime: 11-23-2017 21:04:52
Nov 23 21:08:04 briggs01 xcat: mid05tor12cn16 status: netbooting statustime: 11-23-2017 21:08:04
Nov 23 21:08:16 briggs01 xcat: mid05tor12cn16 status: postbooting statustime: 11-23-2017 21:08:16
Nov 23 21:08:27 briggs01 xcat: mid05tor12cn16 status: booted statustime: 11-23-2017 21:08:27
```

File: perl-xCAT/xCAT/GlobalDef.pm [hunk 1]
```diff
@@ -86,14 +88,15 @@ $::STATUS_BMCREADY     = "bmcready";
...
+#defined->[discovering]->[configuring]->[standingby]->installing->[installed]->booting->[postbooting]->booted->alive,  defined->[discovering]->[configuring]-[standingby]->netbooting->[postbooting]->booted->alive,  alive/unreachable->booting->alive,  powering-off->unreachable, alive->unreachable
 %::NEXT_NODESTAT_VAL = (
     $::STATUS_DEFINED => { $::STATUS_DISCOVERING => 1, $::STATUS_INSTALLING => 1, $::STATUS_NETBOOTING => 1, $::STATUS_POWERING_OFF => 1, $::STATUS_POWERING_ON => 1, $::STATUS_BOOTING => 1, $::STATUS_CONFIGURING => 1 },
     $::STATUS_DISCOVERING => { $::STATUS_INSTALLING => 1, $::STATUS_NETBOOTING => 1, $::STATUS_CONFIGURING => 1, $::STATUS_BOOTING => 1 },
     $::STATUS_CONFIGURING => { $::STATUS_INSTALLING => 1, $::STATUS_NETBOOTING => 1, $::STATUS_STANDING_BY => 1 },
     $::STATUS_INSTALLING => { $::STATUS_INSTALLED => 1, $::STATUS_BOOTING => 1 },
     $::STATUS_INSTALLED => { $::STATUS_BOOTING => 1 },
     $::STATUS_BOOTING => { $::STATUS_BOOTED => 1, $::STATUS_ACTIVE => 1, $::STATUS_INACTIVE => 1 },
+    $::STATUS_POSTBOOTING => { $::STATUS_BOOTED => 1 },
```
Comment (immarvin) [inline perl-xCAT/xCAT/GlobalDef.pm]: should `$::STATUS_POSTBOOTING ` in the value of key `$::STATUS_BOOTING`

Comment (robin2008) [reply perl-xCAT/xCAT/GlobalDef.pm]: Not found there are some function impacted, but I'll add it for `NEXT_NODESTAT_VAL`

**Comment by immarvin** [comment]
agree to merge

**Pull request merged**